megatron/utils.py

```diff
@@ -167,8 +165,11 @@ def get_ltor_masks_and_position_ids(data,
 
 def local_rank():
     """ Local rank of process """
-    return int(os.environ["LOCAL_RANK"])
-
+    local_rank = os.environ.get("LOCAL_RANK")
+    if local_rank is None:
+        print("utils.local_rank() environment variable LOCAL_RANK not set, defaulting to 0", flush=True)
```

Comment:
I think we should default to -1 here? (torch / deepspeed's default if local rank isn't set, i.e we're not distributed.)

Reply:
Some functions rely on rank being equal to 0 (e.g. tensorboard initialization https://github.com/EleutherAI/gpt-neox/blob/319ad174693badc7b1ffde3a2b4b3d068dda588c/megatron/neox_arguments/arguments.py#L106). Would we need to change these functions according to the print_rank_0 function (https://github.com/EleutherAI/gpt-neox/blob/319ad174693badc7b1ffde3a2b4b3d068dda588c/megatron/__init__.py#L17)?